### ISSUE TITLE
`FromHeader` in `AsParameter` does not work with RequestDelegateGenerator

### ISSUE THREAD

**Issue by KennethHoff** [dotnet/aspnetcore#52302]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

If you have a model like the following, it will map erroneously.

```c#
namespace Testing;
public sealed class FormMessage
{
    [FromHeader(Name = "X-Form-Type")]
    public FormType FormType { get; init; } = default;

    [FromBody]
    public IFormContent? Content { get; init; } = default!;
}
```

```c#
var formMessage_local = new global::Testing.FormMessage { X-Form-Type = FormType_local, Content = Content_local };
```

This does work if you use `[FromHeader(Name = "X-Form-Type")]` directly on the handler-parameter itself however.

### Expected Behavior

It would generate this instead:

```c#
var formMessage_local = new global::Testing.FormMessage { FormType = FormType_local, Content = Content_local };
```

### .NET Version

8.0.100

**Comment by martincostello** [comment]
This looks like a duplicate of #51547.

**Comment by KennethHoff** [comment]
Doesn't seem like it was fixed though :§

**Comment by martincostello** [comment]
The fix is scheduled for release in 8.0.1 based on the milestone in #51637.

**Comment by KennethHoff** [comment]
Ah, I see. Sorry then >:)
